File: api/src/models/routine.js
const mongoose = require('mongoose');

const RoutineSchema = new mongoose.Schema({
    _id: { type: String, required: true},
    userId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User',
            required: true,
            index: true
          },
    petId: { type: String,
            ref: 'Pet',
            required: true,
            index: true },
    title: { type: String, required: true, trim:true},
    time: { type: String ,default: ''},
    frequency: { type: String, required: true },
    startDate: { type: Number, required: true },
    endDate: { type: Number, required: true },
    repeatDays: { type: String, default: '' }
}, { timestamps: true });

module.exports = mongoose.model('Routine', RoutineSchema);

File: android/app/src/main/java/com/zahraag/pawsitivehabits/data/models/Routine.kt
package com.zahraag.pawsitivehabits.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
@Entity(tableName = "routine_table")
data class Routine(
    @PrimaryKey @SerializedName("_id")
    val id: String = org.bson.types.ObjectId().toHexString(),
    var userId: String,
    var petId: String,
    var title: String,
    var time: Long?= null,
    var frequency: String,
    var startDate: Long,
    var endDate: Long? =null,
    var repeatDays: String? = null,
    var reminderMinutes: Int = 30,


    @kotlinx.serialization.Transient
    var isSynced: Boolean = false
)
